### PR TITLE
Blog post on v0.2.1 release

### PR DESCRIPTION
**Pull request by tdwils**
Addresses #164 

@AARON-CLARK I have created a rough draft of a new vignette/blog post: "Announcing tidyCDISC 0.2.0". Please take a look and let me know if you have any feedback at this point. I think it still needs some more text and description.

I also added a blog entry to `Blog.Rmd`. 

You mentioned adding links on the `pkgdown` site to the youtube channel. Do you have an idea of where those should go?

Search on your name “AARON” for a couple of questions in the markdown file.

### CODE REVIEW

**Comment by AARON-CLARK** [comment]
FYI @tdwils, we found a pretty big bug with `v0.2.0` in #173 so we'll likely be submitting a patch to CRAN, so I think we'll pivot this "press release" to talk up version `v0.2.1`!

I'll probably spend some time on this this afternoon, so this is an initial list of to-do items for me to follow up on:

1. We always want the latest blog post to show up at the top:

image

3. I don't think we'll broadcast that we're using `OS Health` and `riskmetric` badges since it doesn't really impact the user experience

4. I'm not sure why, but your pngs look a little blurry to me...

image

5. I probably won't include this since it kind of just points out an old bug. Maybe we should include it in the "Miscellaneous enhancements". I'll trim that list a bit.

image

6. I'll update "Contributions on the horizon"

**Comment by Jeff-Thompson12** [comment]
@AARON-CLARK Did you restart R? The JavaScript files don't get automatically updated when running `run_dev()`.

**Comment by Jeff-Thompson12** [comment]
It's not working for me either...

**Comment by tdwils** [comment]
@AARON-CLARK  It doesn't work for me either.

**Comment by Jeff-Thompson12** [comment]
It was my fault. I will be pushing a PR shortly that fixes it.

**Comment by tdwils** [comment]
@AARON-CLARK 

I think we need to update the version of the `fontawesome` package. I get this error when I try to knit the blog post:

image

The circle-dot was added in v4.0.0, and the renv.lock file has 0.2.2.

https://fontawesome.com/icons/circle-dot?s=solid&f=classic

**Comment by Jeff-Thompson12** [comment]
We need to update the renv lock file anyways if we are preparing to submit to CRAN

**Comment by AARON-CLARK** [comment]
> We need to update the renv lock file anyways if we are preparing to submit to CRAN

@Jeff-Thompson12, Is there some other updates to `renv.lock` we need besides `fontawesome` and what's included in #170?

**Comment by AARON-CLARK** [comment]
> @AARON-CLARK
> 
> I think we need to update the version of the `fontawesome` package. I get this error when I try to knit the blog post:
> 
> image
> 
> The circle-dot was added in v4.0.0, and the renv.lock file has 0.2.2.
> 
> https://fontawesome.com/icons/circle-dot?s=solid&f=classic

Actually, it's kind of tricky/redundant to upgrade `fontawesome` and not address the other packages listed here first:

image

@tdwils or @Jeff-Thompson12, could you please take a look at ⁠[this PR](https://github.com/Biogen-Inc/tidyCDISC/pull/170) first and merge if it looks okay?

**Comment by Jeff-Thompson12** [comment]
I guess my thought is that by submitting to CRAN the default is for the most recent versions of the packages to be downloaded by the end user. If we aren't verifying that our application is working as expected on the most recent versions, any new users would be getting a different experience from us.

**Comment by AARON-CLARK** [comment]
> I guess my thought is that by submitting to CRAN the default is for the most recent versions of the packages to be downloaded by the end user. If we aren't verifying that our application is working as expected on the most recent versions, any new users would be getting a different experience from us.

Ahhh, good point. So maybe the our R-CMD-Check for PRs merged into `devel` should be based off of `renv.lock` but our PRs for `master` shouldn't so it checks against the latest versions of our dependencies? Let me know your thoughts and I can adjust #170

**Comment by Jeff-Thompson12** [comment]
I think that approach can help. That combined with a push to make more robust tests that we run locally/on Github but don't submit to CRAN would also be good.

**Comment by AARON-CLARK** [comment]
> I think that approach can help.

Okay, I can make some modifications!

> That combined with a push to make more robust tests that we run locally/on Github but don't submit to CRAN would also be good.

Can you make a new issue for this one?

> Currently, at least one of us should take a test drive of the app with updated packages to make sure everything is still behaving as expected.

I've updated our [Cutting a New Release](https://github.com/Biogen-Inc/tidyCDISC/wiki/Cutting-a-new-release) wiki page accordingly.

**Comment by tdwils** [comment]
From the [Wiki](https://github.com/Biogen-Inc/tidyCDISC/wiki/Cutting-a-new-release#after-merge-into-master):

>Temporarily abandon renv.lock and update all package dependencies to the latest versions (since that's what CRAN will use) to make sure nothing is broken. User needs to test drive all functionality in the app.

Is there a checklist for the functionality we need to test? It would be good to have a standard list of the minimum that needs to be tested.

**Comment by AARON-CLARK** [comment]
> Is there a checklist for the functionality we need to test?

Not Currently!

> It would be good to have a standard list of the minimum that needs to be tested.

Agreed! We could make a separate wiki for that. Basically, we'd want to systematically test each module. We could also set up `shinytest2` tests that perform the desired functionality... but there are some drawbacks to that approach.

**Comment by AARON-CLARK** [comment]
I'm not really sure why `pkgdown` GHA is passing while R-CMD-check is failing with the error below. I would assume `pkgdown::build_site()` would fail if that png was truly missing, but it's not.

image

**Comment by tdwils** [comment]
The file name in .Rbuildignore needs to be updated, and then the test will pass:

https://github.com/Biogen-Inc/tidyCDISC/blob/aaeb9c5e12747b81741091df6caf02a13e324dd2/.Rbuildignore#L22

**Comment by AARON-CLARK** [comment]
> The file name in .Rbuildignore needs to be updated, and then the test will pass

@tdwils, great find!

**Comment by AARON-CLARK** [comment]
Okay @tdwils, `{fontawesome}` has been updated in `renv.lock`. However, I'll keep this PR as DRAFT for now while we consider some changes in #181, which will effect the blogpost's pngs.

**Comment by AARON-CLARK** [comment]
@Jeff-Thompson12, this one is ready for review. Could you please give it a look?

File: renv.lock [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "R": {
-    "Version": "4.2.1",
+    "Version": "4.1.2",
```
Comment (AARON-CLARK) [inline renv.lock:3]: oh @Jeff-Thompson12, is it okay if this stays consistent with RAA? Just so we don't have to cumbersomely change versions back and forth?

Comment (Jeff-Thompson12) [reply renv.lock:3]: That makes sense to me.

File: vignettes/announcing-tidycdisc-0-2-1.Rmd [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+
+- “ALL” is now the default visit selection for standard analysis 41
+
+- Fixed bug when filtering module in Population Explorer where columns for non-selected datasets would display erroneously
+
+- Fixed bug where erroneously variable was listed as a grouping option in the Population Explorer
+
+- Fixed bug accessing files in "app/www/"
+
+- Fixed bug where draggable blocks were not working without inclusion of datasets with ATPT
```
Comment (Jeff-Thompson12) [inline vignettes/announcing-tidycdisc-0-2-1.Rmd:166]: I'm fine with including this note, but the bug was only present when the ATPT dropdown was added to the application. This bug would not have been present in the previous CRAN release.

File: vignettes/announcing-tidycdisc-0-0-1-1.Rmd [hunk 3]
```diff
@@ -462,19 +462,12 @@ In addition to the major features above, this release also upgraded the `pkgdown
...
 
-- **Compare app output with SAS output**. If your goal is to use `tidyCDISC` for visual QC in parallel to your SAS programming efforts, it would be really convenient to compare the app's results to your own without having to sift with your own eye balls. `tidyCDISC` need only allow users to upload SAS results as `sas7bdat` file to generate R's version of a PROC COMPARE.
+So we want to hear from you! Does `tidyCDISC` not work for you? then [tell us why.](https://github.com/Biogen-Inc/tidyCDISC/issues/new) Is there something different about the way you (or your company) handles certain tables / analyses? We want to know about it. Do you have a passion to see clinical trials analysis in R come to fruition in the next few years? Help us shape the process. Do you want to advance the industry forward by contributing to an open source project? Become a `tidyCDISC` developer!
```
[the remark this answers is not in the document]
Comment (Jeff-Thompson12) [reply vignettes/announcing-tidycdisc-0-0-1-1.Rmd:468]: I now see that this mimics the new blog posts conclusion. It's still weird to be editing previous blog posts, but it's not a hill I'm going to die on.

**Review by Jeff-Thompson12 — CHANGES_REQUESTED** [review]
Lots of good work here. I did have a couple comments/conversations to resolve. But they are mostly grammar/wording.

File: README.md [hunk 4]
```diff
@@ -7,8 +7,8 @@
 
 [![R-CMD-check](https://github.com/Biogen-Inc/tidyCDISC/workflows/R-CMD-check/badge.svg)](https://github.com/Biogen-Inc/tidyCDISC/actions)
 [![OS
-Health](https://img.shields.io/badge/OS%20Health-86-green)](https://openpharma.github.io/GithubMetrics/)
-[![riskmetric](https://img.shields.io/badge/riskmetric--0.87-red)](https://pharmar.github.io/riskmetric/)
+Health](https://img.shields.io/badge/OS%20Health-71-yellowgreen)](https://openpharma.github.io/GithubMetrics/)
+[![riskmetric](https://img.shields.io/badge/riskmetric--0.79-red)](https://pharmar.github.io/riskmetric/)
```
[the remark this answers is not in the document]
Comment (AARON-CLARK) [reply README.md:11]: The `render-readme.yaml` GHA actually takes care of that issue. It's only negative when I run it locally. When the GHA renders the `README`, it comes out positive.

File: vignettes/announcing-tidycdisc-0-0-1-1.Rmd (hunk 3 above)
[the remark this answers is not in the document]
Comment (AARON-CLARK) [reply vignettes/announcing-tidycdisc-0-0-1-1.Rmd:468]: LOL. I made an executive decision that it's a bad idea to forecast future work we plan to tackle in our blogposts because there is no guarantee we'll get to it as priorities change. Therefore, I abolished it from old posts.

File: README.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (Jeff-Thompson12) [reply README.md:11]: Why is it negative when selecting the branch on Github then?

[the remark this answers is not in the document]
Comment (AARON-CLARK) [reply README.md:11]: oh, I retract my statement. We don't even have a `render-readme.yaml` for the `tidyCDISC` repo. Sorry, I'm confusing projects. Let me re-inspect!

[the remark this answers is not in the document]
Comment (Jeff-Thompson12) [reply README.md:11]: It would be a great addition. Case in point.

File: vignettes/announcing-tidycdisc-0-2-1.Rmd (hunk 2 above)
Comment (AARON-CLARK) [reply vignettes/announcing-tidycdisc-0-2-1.Rmd:166]: That's true. I think it was available on the previous `master` version (not released on CRAN). I'm fine with removing it.
